unit/util/ssa_expr.cpp

```diff
@@ -48,3 +48,33 @@ TEST_CASE("Set level", "[unit][util][ssa_expr]")
     }
   }
 }
+
+TEST_CASE("Set expression", "[unit][util][ssa_expr]")
+{
+  GIVEN("An SSA expression constructed from a symbol")
+  {
+    const signedbv_typet int_type{32};
+    const array_typet array_type{int_type, from_integer(10, int_type)};
+    const symbol_exprt symbol{"sym", array_type};
+    const index_exprt index{symbol, from_integer(9, int_type)};
+    ssa_exprt ssa(symbol);
+    ssa.set_level_0(1);
+    ssa.set_level_1(3);
+    ssa.set_level_2(7);
+
+    WHEN("call set_expression with an index_exprt")
+    {
+      ssa.set_expression(index);
+      THEN("Indices are preserved")
+      {
+        REQUIRE(ssa.get_level_0() == "1");
+        REQUIRE(ssa.get_level_1() == "3");
+        REQUIRE(ssa.get_level_2() == "7");
+      }
+      THEN("The underlying expression has been replaced")
+      {
+        REQUIRE(ssa.get_original_expr() == index);
+      }
+    }
+  }
+}
```

Comment:
Github complains about a missing newline (which is fixed in "Unit test for the ssa_exprt constructor")